runner.go

```diff
@@ -52,6 +45,31 @@ type doneInfo struct {
 	err error
 }
 
+// RunnerParams holds the parameters for a NewRunner call.
+type RunnerParams struct {
+	// IsFatal is called when a worker exits. If it returns
+	// true, all the other workers
+	// will be stopped and the runner itself will finish.
+	//
+	// If IsFatal is nil, all errors will be treated as fatal.
+	IsFatal func(error) bool
+
+	// When the runner exits because one or more
+	// workers have returned a fatal error, only the most important one,
+	// will be returned. MoreImportant should report whether
+	// err0 is more important than err1.
+	//
+	// If MoreImportant is nil, the first error reported will be
+	// returned.
+	MoreImportant func(err0, err1 error) bool
+
+	// RestartDelay holds the length of time the runner will
+	// wait after a worker has exited with a non-fatal error
+	// before it is restarted.
+	// If this is zero, RestartDelay will be used.
```

Comment:
Perhaps call one of these variables something different? DefaultRestartDelay perhaps?